design/fips-compliance.md

```diff
@@ -0,0 +1,256 @@
+ us FIPS compliancy with minimal work. All that needs to be done is:
+ - Replace the base image from `golang` to the corresponding `goboring/golang
+ `. For example, `golang:1.12` should turn into `goboring/golang:1.12.17b4`
+ - Enable cgo by setting `CGO_ENABLED=1` in the env
+ - Verify that the `authenticator` binary uses the BoringCrypto libraries by
+  running `go tool nm authenticator` and verifying that it has symbols named
+   *_Cfunc__goboringcrypto_*
+   
+More info on the required changes can be found [here](https://github.com/cyberark/conjur-authn-k8s-client/pull/97)
+
+Note: the UX will not change in this solution.
+
+### Non-FIPS compliant image
+
+We should decide if we still want to release a version of the `conjur-authn-k8s-client`
+that is not FIPS compliant (i.e that uses `golang` as the base image). Let's
+look into some pros and cons of both approaches.
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+Note: If the performance of the `goboring/golang` version will be much worse than that
+      of the `golang` version (degradation of more than 5%) then we will release both versions.
```

Comment:
This is a key comment. Is measuring the performance going to be part of this effort then? If so, let us know - we may have some tools we built for secretless that you can leverage.

Reply:
Added a link to the performance section